File: Part1_강의자료/Part1_연습문제_풀이/1764_듣보잡/1764_set.cpp
#include <algorithm>
#include <iostream>
#include <set>
#include <vector>
using namespace std;

int main()
{
    ios::sync_with_stdio(0), cin.tie(0), cout.tie(0);

    int N, M;
    cin >> N >> M;

    string name;
    set<string> unhear;
    for (int i = 0; i < N; i++) {
        cin >> name;
        unhear.insert(name);
    }

    vector<string> ans;
    for (int i = 0; i < M; i++) {
        cin >> name;
        if (unhear.find(name) != unhear.end())
            ans.push_back(name);
    }

    sort(ans.begin(), ans.end());
    cout << ans.size() << '\n';
    for (auto& name : ans)
        cout << name << '\n';

    return 0;
}

File: Part1_강의자료/Part1_연습문제_풀이/1764_듣보잡/1764_set.py
import sys
input = sys.stdin.readline

N, M = map(int, input().split())
unhear = set(input().rstrip() for _ in range(N))

ans = []
for x in [input().rstrip() for _ in range(M)]:
    if x in unhear:
        ans.append(x)

ans.sort()
print(len(ans))
for name in ans:
    print(name)
